c++/include/orc/orc-config.hh.in

```diff
@@ -75,4 +75,38 @@
   }
 #endif
 
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+  namespace orc {
+    /**
+     * Compute value * repetitions, return false if overflow, return true otherwise
+     * and save the result at the address pointed to by result
+     * imitates the jdk Math.multiplyExact implementation
+     * but this method makes the assumption that repetitions > 1
+     */
+    static bool multiplyExact(int64_t value, int64_t repetitions, int64_t* result) {
+      int64_t r = value * repetitions;
+      if (((value < 0 ? -value : value) | repetitions) >> 31 != 0 && r / repetitions != value) {
+        return false;
+      }
+      *result = r;
+      return true;
+    }
```

Comment:
Hi @dongjoon-hyun @wgtmac. Finally passed the test. But it needs a strict review.
The window environment overflow check I've mimicked the JDK implementation. If there is a better way, please let me know.
In fact it doesn't work well on clang (the compiler has some unknown behaviour that I don't know about)

Reply:
`#ifdef _MSC_VER` is enough to assert that it is on Windows. You may also leverage cmake_modules/CheckSourceCompiles.cmake to figure out if the gcc builtin functions are available or not. BTW, orc-config.hh.in is to address capability of older c++ standards. If would be better to handle these kinds of functions in Adaptor.hh.in.